docs/rfcs/0013-export-deployment-traces-with-otlp.md

```diff
@@ -0,0 +1,94 @@
+- Start Date: 2024-06-13
+- Target Version: 0.49.0
+
+# Summary
+
+This RFC proposes a new feature to export the OpenTelemetry Trace with deployment and stage spans.
+
+# Motivation
+
+We want to investigate deployment performances, such as the time to complete each deployment or stage.
+We have `deployment_status` metrics for now, but this metric has one metric for each deployment status, and the count of metrics variety only glows larger.
+When we save all histories of such metrics, we need more storage as time goes on since the completed deployments have a 0/1 value for each status.
+Conversely, if we save the histories only while it's running, we have to fill the metrics with the last value or zero to get some statistics for deployments.
+
+So, I propose to separate the metrics to get some statistics and the traces to get deployment performances.
+This RFC proposes the latter, the traces, to get deployment performances.
+
+# Detailed design
+
+## Architecture
+
+Collect spans at piped
+→ send to pipecd-gateway envoy
+→ proxies to OpenTelemetry Collector
+→ send anywhere with exporters
+
```

Comment:
[ask]
This is just curious.
In this RFC, do you consider the case when users want to use their own otel collector?

Reply:
If you mean 'users' is control plane managers, the answer is 'Yes.'
If you mean 'users' is piped managers, the answer is 'No.'

When control plane managers want to send traces to SaaS, the collector must include exporters to send them. They may need to select the collector image and configure it.

Reply:
> If you mean 'users' is control plane managers, the answer is 'Yes.'
If you mean 'users' is piped managers, the answer is 'No.'

OK, I got your point!
So it might be nice to add an explanation that this is for the control plane managers.

Reply:
I thought some 'piped managers' wanted to check the traces and send their own maintained otel-collector to check the deployment.

Reply:
I added mention of users this RFC targets.
https://github.com/pipe-cd/pipecd/pull/4971/commits/c59f4f857ffff3b16e92bec4cb17663251f5d70a

Reply:
> I thought some 'piped managers' wanted to check the traces and send their own maintained otel-collector to check the deployment.

I agree, but I want to implement the method for control plane managers first.